src/pages/ProductListing/ProductListing.js

```diff
@@ -0,0 +1,49 @@
+import axios from "axios";
+import { useEffect, useState } from "react";
+import { useProductList } from "../../context/ProductListContext";
+import { Filters_view, Product_card, Mobile_filter } from "../../Components";
+import { useFilter } from "../../context/FilterContext";
+import "./ProductListing.css";
+import LoadingAnimation from "react-circle-loading-animation";
+
+import { filteration } from "../../utils/filteration";
+export const ProductListing = () => {
+  const { product_list, setProductList } = useProductList();
+  const [isloading, setisloading] = useState(false);
+  const { filter_state } = useFilter();
+  useEffect(() => {
```

Comment:
you can make custom hook for api call and extract into separate file so it will looks cleaner and reusable